Apigee/ManagementAPI/DeveloperAppAnalytics.php

```diff
@@ -152,20 +152,27 @@ public function getAllEnvironments()
      *    A comma separated list of the same values as $metric.
      * @param string $sortOrder
      *    Either 'ASC' or 'DESC'.
+     * @param bool $is_company
+     *    If TRUE get app for company, otherwise for developer.
      *
      * @return array
+     *   An array of analytic data points.
+     *
+     * @throws \Apigee\Exceptions\ParameterException
+     *   Thrown in case of an invalid parameter.
      */
-    public function getByAppName($devIdOrCompanyName, $appName, $metric, $tStart, $tEnd, $tUnit, $sortBy, $sortOrder = 'ASC')
+    public function getByAppName($devEmailOrCompany, $appName, $metric, $tStart, $tEnd, $tUnit, $sortBy, $sortOrder = 'ASC', $is_company = FALSE)
     {
         $params = self::validateParameters($metric, $tStart, $tEnd, $tUnit, $sortBy, $sortOrder);
 
         // We need to filter analytics by the developer or company name. If we do not
         // we will get back data for all apps with the app name, since they are not
         // unique.  For example, two developers can make an app named "test".
-        if (!empty($devIdOrCompanyName)) {
-            $org = $this->config->orgName;
-            $params['filter'] = "(developer eq '$org@@@$devIdOrCompanyName')";
+        if (!empty($devEmailOrCompany)) {
```

Comment:
`devEmailOrCompany` is a required parameter, it should not be empty. In case if `` is provided as a parameter value it must be user error.